src/core/conversation.py

```diff
@@ -100,7 +111,31 @@ def perform_next_step(self) -> Message:
         else:
             raise Exception(f"Invalid conversation role: {last_message.role}")
 
-        return self._get_last_message()
+        last_message = self._get_last_message()
+        error_limit = 5
+        error_count = 0
+        while (
+            last_message.role == ConversationRolesInternalEnum.CODE
+            and "Traceback" in last_message.content
+        ):
+            # Ask code assistant to fix the error and overwrite previous output. Do until error is fixed.
+
+            error_count += 1
+            if error_count >= error_limit:
+                raise Exception("Too many errors in code.")
```

Comment:
IMO we should be defining our own exception instead on using default one

Reply:
Okay, will do